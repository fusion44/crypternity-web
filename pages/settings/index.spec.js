import { shallow } from "@vue/test-utils"
import index from "./index"

describe("Settings Index", () => {
  it("renders the correct markup default state", () => {
    let wrapper = shallow(index)

    let html = wrapper.html()
    expect(html).toMatchSnapshot()
  })

  it("renders the correct markup with admin user", () => {
    const wrapper = shallow(index)
    wrapper.setData({ isAdmin: true })

    expect(wrapper.vm.isAdmin).toBe(true)

    let html = wrapper.html()
    expect(html).toMatchSnapshot()
  })
})
