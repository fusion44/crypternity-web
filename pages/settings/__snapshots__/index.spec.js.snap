// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Index renders the correct markup default state 1`] = `
<div>
  <v-tabs dark="dark">
    <v-tab>User Data</v-tab>
    <v-tab-item>
      <v-card flat="flat">
        <h1>Update My Account Settings</h1>
      </v-card>
    </v-tab-item>
    <v-tab>System Admin</v-tab>
    <v-tab-item>
      <v-card flat="flat">
        <h3>Update Supported Coins </h3>
        <v-btn disabled="true">Update</v-btn>
      </v-card>
    </v-tab-item>
  </v-tabs>
</div>
`;

exports[`Settings Index renders the correct markup with admin user 1`] = `
<div>
  <v-tabs dark="dark">
    <v-tab>User Data</v-tab>
    <v-tab-item>
      <v-card flat="flat">
        <h1>Update My Account Settings</h1>
      </v-card>
    </v-tab-item>
    <v-tab>System Admin</v-tab>
    <v-tab-item>
      <v-card flat="flat">
        <h3>Update Supported Coins </h3>
        <v-btn>Update</v-btn>
      </v-card>
    </v-tab-item>
  </v-tabs>
</div>
`;
